pkg/foundation/semaphore/semaphore.go

```diff
@@ -31,7 +31,7 @@ type Simple struct {
 // Ticket reserves a place in the queue and can be used to acquire access to a
 // Lock.
 type Ticket struct {
-	Lock
+	l Lock
```

Comment:
This makes the semaphore use a bit safer, previously a caller could do `Ticket.Lock` to get the lock object without actually acquiring it through the semaphore. Of course that is an invalid use of the struct, but now we prevent that entirely.